core/src/main/java/com/datastrato/gravitino/catalog/TableOperationDispatcher.java

```diff
@@ -83,21 +83,22 @@ public NameIdentifier[] listTables(Namespace namespace) throws NoSuchSchemaExcep
   @Override
   public Table loadTable(NameIdentifier ident) throws NoSuchTableException {
     EntityCombinedTable table =
-        TreeLockUtils.doWithTreeLock(ident, LockType.READ, () -> loadCombinedTable(ident));
-
-    if (table.imported()) {
-      return table;
-    }
-
-    if (GravitinoEnv.getInstance()
-        .schemaDispatcher()
-        .schemaExists(NameIdentifier.of(ident.namespace().levels()))) {
-      EntityCombinedTable importedTable =
-          TreeLockUtils.doWithTreeLock(
-              NameIdentifier.of(ident.namespace().levels()),
-              LockType.WRITE,
-              () -> importTable(ident));
-      return importedTable;
+        TreeLockUtils.doWithTreeLock(ident, LockType.READ, () -> internalLoadTable(ident));
+
+    if (!table.imported()) {
+      // Load the schema to make sure the schema is imported.
+      SchemaDispatcher schemaDispatcher = GravitinoEnv.getInstance().schemaDispatcher();
+      NameIdentifier schemaIdent = NameIdentifier.of(ident.namespace().levels());
+      schemaDispatcher.loadSchema(schemaIdent);
```

Comment:
It will produce an extra event of `loadSchema` if we use the loadSchema instead of `SchemaExists`.  Is it expected?

Reply:
But using `SchemaExists` seems a bit weird...